libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1601,6 +1601,75 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_insert_edge_from_coordinates(int meshKernelId,
+                                                                double firstNodeX,
+                                                                double firstNodeY,
+                                                                double secondNodeX,
+                                                                double secondNodeY,
+                                                                int& firstNodeIndex,
+                                                                int& secondNodeIndex,
+                                                                int& edgeIndex)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+
+            std::unique_ptr<meshkernel::CompoundUndoAction> compoundUndoAction = meshkernel::CompoundUndoAction::Create();
+            meshkernel::Point const firstNodeCoordinates{firstNodeX, firstNodeY};
+            meshkernel::Point const secondNodeCoordinates{secondNodeX, secondNodeY};
+
+            const auto newEdgeLength = ComputeDistance(firstNodeCoordinates, secondNodeCoordinates, meshKernelState[meshKernelId].m_projection);
+            const auto edgeLengths = meshKernelState[meshKernelId].m_mesh2d->m_edgeLengths;
+            const auto lengthFraction = 0.01;
```

Comment:
Perhaps make edgeLengths a reference
Unless mesh->m_edgeLengths is modified later and you do not want t get these updates.